conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/RawTypes.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.conjure.java.client.jaxrs.feignimpl;
+
+import com.google.common.reflect.TypeToken;
+import java.lang.reflect.Type;
+
+/**
+ * Utility functionality to detect raw types.
+ */
+final class RawTypes {
+
+    /** Attempts to extract the raw type class from a {@link Type}. */
+    static Class<?> get(final Type type) {
+        return TypeToken.of(type).getRawType();
+    }
+
+    private RawTypes() { }
+}
```

Comment:
This class is new

Reply:
Do you think there's any realistic danger that the old `feign.Types#getRawType` method could return different results than guava's TypeToken class?

I think this repo's testing is limited to pretty vanilla usages unfortunately... just wondering if this is going to break some strange service out there...

Reply:
I have more confidence in the google implementation than `feign.Types`. I've run OpenFeign tests with their implementation replaced with this one without failures.